BuilderDesignPattern/src/main/java/com/designPattern2/Flat.java

```diff
@@ -8,7 +8,7 @@ public class Flat {
     private final boolean garden;
 
     public Flat(BuildingType buildingType, int bedrooms, int bathrooms, boolean garden) {

```

Comment:
I like that you changed this to ensure that the type is always `BuildingType.FLAT` - although now you have an unused parameter in this constructor that you can remove